msg.go

```diff
@@ -369,6 +370,15 @@ Loop:
 					}
 				}
 			}
+			// never exceed the allowed label count lenght (63) 
+			if labels >= 63 { 
+				return "", lenmsg, &Error{err: "name exceeds 63 labels"}
+			}
+			labels += 1
+			// never exceed the allowed doman name length (255 octets) 
+			if len(s) >= 255 {
```

Comment:
This is erroneously checking presentation-format length, but the [RFC 1035 limit](https://tools.ietf.org/html/rfc1035#section-2.3.4) applies to _wire_ format. As a result, it rejects the following perfectly valid (if unusual) domain name:
```
\.\(\)\;\.\(\)\;\.\(\)\;.32-octets-including-length--001.32-octets-including-length--002.32-octets-including-length--003.32-octets-including-length--004.32-octets-including-length--005.32-octets-including-length--006.32-octets-including-length--007.13-octet-TLD.
```

Also, note that 255 octets is an acceptable length; it's 256 that is invalid.